### PR TITLE
Adding Passio Nutrition AI agent

### PR DESCRIPTION
**Pull request by ivyas21**
# Description

Add Passio Nutrition AI Food Search Tool to LLAMA_index

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

### CODE REVIEW

**Comment by ivyas21** [comment]
@kells1986, @drstarson 
Details: https://www.loom.com/share/d9fc6575200a44b1ae03ea213b5e26f3

File: llama-index-integrations/tools/llama-index-tools-passio-nutrition-ai/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Passio Nutrition AI Tool
+
+This tool connects to a Passio Nutrition AI account and allows an Agent to perform searches for 4000 different unique foods, 1.5M packaged food, images and videos.
```
Comment (kells1986) [inline llama-index-integrations/tools/llama-index-tools-passio-nutrition-ai/README.md]: I would change this to something along the lines of "searches against a database of over 2.2M foods". 

The 4000 number is the number of visual labels, they are then mapped to ~40k nutrition references, and then we have 2.2M packaged food records in elastic. 

I don't think the number 4000 will be relevant for users of the tool. We also aren't providing images or videos over the API.

File: llama-index-integrations/tools/llama-index-tools-passio-nutrition-ai/examples/passio_nutrition_ai.ipynb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      "========================\n",
+      "\n",
+      "The slice of pepperoni pizza you were eating contains approximately 2523.9 calories. It's important to note that this calorie count is for a 14\" pizza with pepperoni topping and regular crust. If you are looking for a healthier option, you may consider alternatives with fewer calories.\n"
```
Comment (kells1986) [inline llama-index-integrations/tools/llama-index-tools-passio-nutrition-ai/examples/passio_nutrition_ai.ipynb]: >The slice of pepperoni pizza you were eating contains approximately 2523.9 calories

I think we might need to change that example 😅 It's taken a whole (huge) pizza. Let's see if we can get an example with a good answer into the notebook

**Comment by kells1986** [comment]
Looks good Ishani, I added a couple of minor comments

File: llama-index-integrations/tools/llama-index-tools-passio-nutrition-ai/examples/passio_nutrition_ai.ipynb (hunk 2 above)
Comment (ivyas21) [reply llama-index-integrations/tools/llama-index-tools-passio-nutrition-ai/examples/passio_nutrition_ai.ipynb]: Yes, I'll change it for sure. I realized it once I push it.

File: llama-index-integrations/tools/llama-index-tools-passio-nutrition-ai/README.md (hunk 1 above)
Comment (ivyas21) [reply llama-index-integrations/tools/llama-index-tools-passio-nutrition-ai/README.md]: Thank you. I modified it.

**Comment by ivyas21** [comment]
@kells1986  I applied all changes. I am merging PR to master.

**Pull request merged**